climada/entity/exposures/base.py

```diff
@@ -678,13 +678,15 @@ def read_hdf5(self, file_name):
         """
         LOGGER.info('Reading %s', file_name)
         with pd.HDFStore(file_name) as store:
-            self.__init__(store['exposures'])
             metadata = store.get_storer('exposures').attrs.metadata
+            # in previous versions of CLIMADA and/or geopandas, the CRS was stored in '_crs'/'crs'
+            crs = metadata.get('crs', metadata.get('_crs'))
```

Comment:
I do not really understand this line, which seems to be the crucial one.

Reply:
The CRS used to be stored as an explicit attribute. In geopandas previous to 0.8, this attribute was called `crs`, and starting from 0.8, it was called `_crs`. With the introduction of the `gdf` attribute in CLIMADA, CLIMADA doesn't store the GeoDataFrame's attributes in the HDF file, but uses its own `meta` attribute to store and load the CRS.

This line makes sure that an explicitly stored `crs`/`_crs` attribute (from an old file) has precedence over the CRS stored in the `meta` attribute. Only if there is no `crs`/`_crs` attribute or if the attribute's value is `None`, the CRS stored in `meta` is used.

Reply:
Thanks for the explanation! So, just to be sure, no error would arise is CRS is neither of the 3 locations?

Reply:
Exactly, in that case, the local `crs` variable would get the value `None` and the would be passed on to `__init__` which then has a default way of how to deal with this case (it will default to EPSG:4326).

Reply:
Great!